smartcontractkit/external-adapters-js
# [SC-47782] Master branch changeset check fails

## Closes SC-47782

## Description

Removed duplicate adapter-changeset validation from Test & Build workflow. The Test & Build workflow triggered this validation on master which would fail every time since changesets are consumed on develop. This action is already triggered under its own workflow and properly configured to avoid validating on master. 

## Changes

- Removed adapter-change-has-changeset from Test & Build workflow

## Steps to Test

1. Ensure "Adapter changes accompanied by a changeset / Adapter changes accompanied by a changeset" runs on push/pull requests for develop
2. Ensure "Test & build / Adapter changes accompanied by a changeset" does not run anymore
3. Ensure the changeset validation does not run at all for PR's on master

*Note: Unit and integration tests are not needed since this is a change to GitHub workflows

## Quality Assurance
